File: app/src/main/java/com/ocram/qichwadic/framework/net/client/RetrofitClient.java
package com.ocram.qichwadic.framework.net.client;

import com.ocram.qichwadic.domain.model.Definition;
import com.ocram.qichwadic.domain.model.Dictionary;
import com.ocram.qichwadic.domain.model.SearchResult;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Flowable;

public class RetrofitClient implements Service {

    private RetrofitService retrofitService;

    @Inject
    public RetrofitClient(RetrofitService retrofitService) {
        this.retrofitService = retrofitService;
    }

    @Override
    public Flowable<List<Dictionary>> getDictionaries() {
        return this.retrofitService.getDictionaries(1);
    }

    @Override
    public Flowable<List<SearchResult>> searchWord(String searchWord, int fromQuechua, String target, int searchType) {
        return retrofitService.searchWord(1, searchWord, fromQuechua, target, searchType);
    }

    @Override
    public Flowable<List<Definition>> fetchMoreResults(int dictionaryId, String word, int fromQuechua, int searchType, int page) {
        return this.retrofitService.fetchMoreResults(dictionaryId, word, fromQuechua, searchType, page);
    }

    @Override
    public Flowable<List<Definition>> getAllDefinitionsByDictionary(int dictionaryId) {
        return this.retrofitService.getAllDefinitionsByDictionary(dictionaryId);
    }
}

File: app/src/main/java/com/ocram/qichwadic/features/common/data/remote/RetrofitClient.kt
package com.ocram.qichwadic.features.common.data.remote

import com.ocram.qichwadic.features.common.data.model.DefinitionEntity
import com.ocram.qichwadic.features.common.data.model.DictionaryEntity
import com.ocram.qichwadic.features.common.data.model.SearchResultEntity
import retrofit2.Response

class RetrofitClient(private val retrofitService: RetrofitService) : Service {

    override suspend fun getDictionaries(): Response<List<DictionaryEntity>> {
        return this.retrofitService.getDictionaries(1)
    }

    override suspend fun searchWord(searchWord: String, fromQuechua: Int, target: String, searchType: Int): Response<List<SearchResultEntity>> {
        return retrofitService.searchWord(1, searchWord, fromQuechua, target, searchType)
    }

    override suspend fun fetchMoreResults(dictionaryId: Int, word: String, fromQuechua: Int, searchType: Int, page: Int): Response<SearchResultEntity> {
        return this.retrofitService.fetchMoreResults(dictionaryId, word, fromQuechua, searchType, page)
    }

    override suspend fun getAllDefinitionsByDictionary(dictionaryId: Int): Response<List<DefinitionEntity>> {
        return this.retrofitService.getAllDefinitionsByDictionary(dictionaryId)
    }
}
